### ISSUE TITLE
An issue from Fraunhofer-Chalmers Centre: Compute Capability 3.0

### ISSUE THREAD

**Issue by bnase** [NVIDIA/AMGX#8]
Is CC 3.0 really not compatible? We have tested and everything seems to be fine on 660ti. Nonetheless, there are many occurrences of hardcoded requirement of CC>=3.5. Is it safe to change to >=3.0?

**Comment by marsaev** [comment]
You are right, we mainly used switch CUDA_ARCH with 3.5 to understand whether we can use warp functions. Considering that CC < 3.0 is not supported anymore i would say it's safe to drop those codepaths.
In some cases (i.e. loading global memory) code is based on what was the best for kepler GPUs, which might be a little bit outdated now, and it would make sense to add new branches for more optimal options for newer CCs.

**Comment by marsaev** [comment]
Tracking internally: AMGX-57
